packages/flutter_tools/lib/executable.dart

```diff
@@ -179,8 +179,11 @@ Future<Null> _exit(int code) async {
   logger.flush();
 
   // Give the task / timer queue one cycle through before we hard exit.
-  await Timer.run(() {
+  Completer<Null> completer = new Completer<Null>();
+  Timer.run(() {
```

Comment:
Using a Timer will mean that futures with completion handlers that themselves await things will have time to finish too. Using a Future would mean that we don't drain all the microtasks, only those queued before this one.
Sounds like what you're saying is that I should just remove the await, complete the _exit early (rather than never, as now), and add a comment explaining all this.